trunk/NDHMS/OrchestratorLayer/config.f90

```diff
@@ -0,0 +1,924 @@
+          SUBRTSWCRT,OVRTSWCRT,AGGFACTRT, &
+          GWBASESWCRT,  GW_RESTART,RSTRT_SWC,TERADJ_SOLAR, &
+          sys_cpl, gwChanCondSw, GwPreCycles, GwSpinCycles, GwPreDiagInterval, &
+          gwsoilcpl, UDMP_OPT
+     logical:: GwPreDiag, GwSpinUp
+     real:: DTRT_TER,DTRT_CH, DTCT, dxrt0,  gwChanCondConstIn, gwChanCondConstOut, gwIhShift
+     character(len=256) :: route_topo_f=""
+     character(len=256) :: route_chan_f=""
+     character(len=256) :: route_link_f=""
+     character(len=256) :: route_lake_f=""
+     character(len=256) :: route_direction_f=""
+     character(len=256) :: route_order_f=""
+     character(len=256) :: gwbasmskfil =""
+     character(len=256) :: gwstrmfil =""
+     character(len=256) :: geo_finegrid_flnm =""
+     character(len=256) :: udmap_file =""
+     character(len=256) :: GWBUCKPARM_file = ""
+     integer :: reservoir_data_ingest ! STUB FOR USE OF REALTIME RESERVOIR DISCHARGE DATA. CURRENTLY NOT IN USE. 
+     character(len=1024) :: reservoir_obs_dir = ""
+
+     logical :: compound_channel
+     integer ::frxst_pts_out            ! ASCII point timeseries output at user specified points
+     integer ::CHRTOUT_DOMAIN           ! Netcdf point timeseries output at all channel points
+     integer ::CHRTOUT_GRID                ! Netcdf grid of channel streamflow values
+     integer ::CHANOBS_DOMAIN             ! NetCDF point timeseries of output at forecast/gage points
+     integer ::LSMOUT_DOMAIN              ! Netcdf grid of variables passed between LSM and routing components
+     integer ::RTOUT_DOMAIN                ! Netcdf grid of terrain routing variables on routing grid
+     integer ::output_gw                   ! Netcdf grid of GW
+     integer ::outlake                   ! Netcdf grid of lake
+     integer :: rtFlag
+     integer ::khour
+
+     character(len=256) :: nudgingParamFile
+     character(len=256) :: netwkReExFile
+     logical            :: readTimesliceParallel
+     logical            :: temporalPersistence
+     logical            :: persistBias
+     logical            :: biasWindowBeforeT0
+     character(len=256) :: nudgingLastObsFile
+     integer            :: minNumPairsBiasPersist
+     integer            :: maxAgePairsBiasPersist
+     logical            :: invDistTimeWeightBias
+     logical            :: noConstInterfBias  
+     character(len=256) :: timeSlicePath
+     integer            :: nLastObs
+     integer            :: bucket_loss
+
+   contains
+
+     procedure, pass(self) :: check => rt_nlst_check
+     
+  END TYPE namelist_rt_
+
+  type, public :: Configuration_
+   contains
+     procedure, nopass :: init => config_init
+     procedure, nopass :: init_nlst => init_namelist_rt_field
+  end type Configuration_
+
+  integer, parameter :: max_domain = 5
+
+  type(NOAHLSM_OFFLINE_), protected, save :: noah_lsm
+  type(WRF_HYDRO_OFFLINE_), protected, save :: wrf_hydro
+  type(namelist_rt_), dimension(max_domain), save :: nlst
```

Comment:
Should this be `protected`? Why not? Ideally, this should be read-only.

Reply:
Yes, I agree that it should be read-only. At the moment, DTCT and sincedate are read from file using netcdf in module_NWM_io.F. DTCT is also manipulated with conditions like the following:
nlst(did)%DTCT = min(nlst(did)%DTCT, nlst(did)%DTRT_CH).
What do you recommend? Creating a private copy of the namelist, moving these changes into the config.f90?